lib/drivers/abbott/freeStyleLibreData.js

```diff
@@ -233,33 +233,59 @@ export default class FreeStyleLibreData {
   }
 
   buildTimeChangeRecords() {
-    this.records.filter(elem =>
-      (elem.headerFields.recordType === DB_RECORD_TYPE.TIME_CHANGE_RESULT)
-      || (elem.headerFields.recordType === DB_RECORD_TYPE.USER_TIME_CHANGE))
-      .forEach((record) => {
-        const oldDateTime = this.getDateTime(
-          record.timeChangeFields.oldReaderTime,
-          record.timeChangeFields.oldUserTimeOffset,
-        );
-
-        const timeChange = this.cfg.builder.makeDeviceEventTimeChange()
-          .with_change({
-            from: sundial.formatDeviceTime(oldDateTime),
-            to: sundial.formatDeviceTime(record.jsDate),
-            agent: 'manual',
-          })
-          .with_deviceTime(sundial.formatDeviceTime(record.jsDate))
-          .set('index', record.headerFields.recordNumber)
-          .set('jsDate', record.jsDate);
-
-        // check if this time change is a duplicate of the previous one from a different DB
-        const previousRecord = this.postRecords[this.postRecords.length - 1];
-        if (!(previousRecord
-          && previousRecord.subType === 'timeChange'
-          && _.isEqual(previousRecord.change, timeChange.change))) {
-          this.postRecords.push(timeChange);
+    let timeChanges = this.records.filter(elem =>
+      (elem.headerFields.recordType === DB_RECORD_TYPE.TIME_CHANGE_RESULT) ||
+      (elem.headerFields.recordType === DB_RECORD_TYPE.USER_TIME_CHANGE));
+
+    // because the time changes exist in multiple DBs there will be duplicates
+    timeChanges = _.uniqBy(timeChanges, o =>
+      o.headerFields.readerTime + o.headerFields.userTimeOffset);
+
+    for (let i = 0; i < timeChanges.length; i++) {
+      let record = timeChanges[i];
+
+      const oldDateTime = this.getDateTime(
+        record.timeChangeFields.oldReaderTime,
+        record.timeChangeFields.oldUserTimeOffset,
+      );
+
+      const nextChange = timeChanges[i + 1];
+
+      if (nextChange) {
+        const distance = nextChange.headerFields.recordNumber - record.headerFields.recordNumber;
+        // even if combined date/time changes are made directly after another,
+        // they should still be more than 50 records apart
```

Comment:
Where does this magic number come from? Is that defined in a spec? As a maintainer, if this came from a spec, and the spec was wrong, I would take a different action than trying a different magic number :)

Reply:
Nope, not from spec, but from observation. Which is why I defined it as a constant in case we want to change it in future. I wish there was a better way of doing this, but since the spec does not distinguish between date changes and time changes (but still records them separately), I resolved to just looking if there are date/time change events near each other.

We follow a similar approach in the Tandem driver:

https://github.com/tidepool-org/chrome-uploader/blob/develop/lib/drivers/tandem/tandemDriver.js#L1492-L1495

Reply:
Understood. It’s a bit disconcerting how much trial and error is involved in driver development.

Reply:
Understood. It’s a bit disconcerting how much trial and error is involved in driver development.